tests/ci/cdk/run-cdk.sh

```diff
@@ -215,6 +215,27 @@ function setup_ci() {
 
   create_github_ci_stack
   create_android_resources
+  create_m1_ec2_instance
+}
+
+function create_m1_ec2_instance() {
+  # Use aws cli to create an M1 ec2 instance.
+  # TODO: Move resource creation to aws cdk when cdk has support for M1 ec2 instances.
+  # Issue: https://github.com/aws/aws-cdk/issues/23854
+  # AMI is for M1 MacOS Monterey.
```

Comment:
Can this use a raw cloudformation construct like we do for the fuzzing? https://github.com/awslabs/aws-lc/blob/main/tests/ci/cdk/cdk/aws_lc_github_fuzz_ci_stack.py#L112-L125

Reply:
Apparently `CfnInstance` exists, so I've used that instead to launch the ec2 instance in cdk.
* https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_ec2/CfnInstance.html